src/navmazing/_navigator.py

```diff
@@ -65,11 +65,18 @@ def navigate(
                 stacklevel=2,
             )
 
-        if args or kwargs:
+        allowed = ["use_resetter", "wait_for_view"]
+        if (args and not set(args).issubset(allowed)) or (
+            kwargs and not set(kwargs.keys()).issubset(allowed)
+        ):
             args_str = ", ".join(
                 chain(
-                    map(repr, args),
-                    (f"{key}={value!r}" for key, value in kwargs.items()),
+                    map(repr, [a for a in args if a not in allowed]),
```

Comment:
Just use kwonly here as well

Reply:
want to make sure I understand your suggestion as I'm reading two things.
1. args check here unnecessary
2. add use_resetter and wait_for_view to `navigate` function signature to force kwonly

I actually had a commit with kwonly for the function but threw it out because it didn't seem right. Can see how that will actually be right - and will prevent this warning too.

Reply:
the parameters used here should be kwonly to manage passover without affecting the affecting of the deprecated args/kwargs

the warning should still continue as arbitrary navigate arguments will be removed soon